File: TypeScript/src/players.ts
export class Players {

    elements = [];

    public add(playerName: string): void {
        this.elements.push(playerName);
    }

    public println(): void {
        this.printlnAdded();
        this.printlnNum();
    }

    private printlnAdded(): void {
        console.log(this.elements[this.elements.length - 1] + " was added");
    }

    private printlnNum(): void {
        console.log("They are player number " + this.count());
    }

    public count(): number {
        return this.elements.length;
    }

}

File: Csharp/Trivia/Players.cs
using System;
using System.Collections.Generic;

namespace UglyTrivia
{
    public class Players
    {
        internal List<String> elements = new List<String>();

        public void Add(String playerName)
        {
            elements.Add(playerName);
        }

        public virtual void Println()
        {
            PrintlnAdded();
            PrintlnNum();
        }

        private void PrintlnAdded()
        {
            Console.WriteLine(elements[elements.Count - 1] + " was added");
        }

        private void PrintlnNum()
        {
            Console.WriteLine("They are player number " + Count());
        }

        public int Count()
        {
            return elements.Count;
        }
    }
}
